Dockerfile.rhel

```diff
@@ -0,0 +1,22 @@
+# This Dockerfile is used by CI to publish odo binary artifacts.
+# It builds an image containing the Mac, Win and Linux version of odo binary on the
+# OpenShift golang image.
+
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.11 AS builder
+LABEL Description="This image is used to build odo(Developer-focused CLI for OpenShift) \
+executables for Windows, macOS and linux" Version="1.0"
+
+COPY . /go/src/github.com/openshift/odo
+WORKDIR /go/src/github.com/openshift/odo
+RUN go get github.com/mitchellh/gox &&\
+    make cross
+
+# For local testing use registry image registry.access.redhat.com/ubi7/ubi instead of
+# registry.svc.ci.openshift.org/ocp/4.2:cli
+FROM registry.svc.ci.openshift.org/ocp/4.2:cli
```

Comment:
Why are we using cli image? and not ubi?

Reply:
> Why are we using cli image why not just registry.svc.ci.openshift.org/ocp/4.2:base or ubi?

I am referring the same format used in [1]. Even i don't know what will be the side effect on ocp release team if ```registry.svc.ci.openshift.org/ocp/4.2:base``` or ```ubi`` is used. Lets wait how @jwforres sees the change. cc_ @mohammedzee1000  

[1] https://github.com/openshift/origin/blob/master/images/cli-artifacts/Dockerfile.rhel

Reply:
This is to provide the odo artifacts in the image, based on the openshift one. I expect this to be one of our primary deliverables to release

Reply:
yes, why do we need to base this on openshift CLI image if we don't need oc?

Reply:
@kadel is right. So @mohammedzee1000 its worth having to use ```ubi``` image, isn't it ?

Reply:
Well its what the image that openshift uses to host alltheir cli artifacts. not just oc. Imight not be the best person to answer the reason why @sspeiche @jwforres

Reply:
Ping @sspeiche @jwforres

Reply:
If there is no specific reason why use this, I would rather use UBI

Reply:
Done